Create SPARQL-JV.md

jonathanvajda opened the pull request:
This is my attempt. I have only a handful that need their SPARQL queries fixed (they are obviously broken). Those are marked with "borked" in the body.

I will adjust those in next 24 hrs. Any PRs welcome.

jonathanvajda committed:
Create SPARQL-JV.md

This is my attempt. I have only a handful that need their SPARQL queries fixed (they are obviously broken). Those are marked with "borked" in the body.

I will adjust those in next 24 hrs. Any PRs welcome.

jonathanvajda committed:
updated sparql queries

Tweaked 7-1, 7-3
Fixed 7-8
Realized 7-9 is actually more difficult, and a Kata-6 (now Problem 6-7)

Still have to fix:
6-3, 6-6
5-3, 5-4, 5-5
4-2, 4-4

Problem 4-2 should probably be harder. I have a plan.

jonathanvajda committed:
Update SPARQL-JV.md

Tweaked 7-7
Fixed 6-3
Fixed 6-6

Remaining:
5-3, 5-4, 5-5
4-2, 4-4

Problem 4-2 should probably be harder. I have a plan.

jonathanvajda committed:
Update SPARQL-JV.md

fixed code format

jonathanvajda committed:
Update SPARQL-JV.md

Fixed 5-3
Swapped 4-2 (became 5-4) and 5-4 (became 4-2)
Fixed 5-4
Fixed 4-2

Still need to fix 5-5 and 4-4

jonathanvajda committed:
updates to a pile of queries

need to fix 5-5 and 4-4

jonathanvajda committed:
updated 5-5

I am not confident that the syntax is right for the negation of the regex. I am going for a negation of a disjunction

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
Update SPARQL-JV.md

jonathanvajda committed:
4-1 fixed

It was so janky and the difficulty wasn't up to Kata-4, imo. I think it is sufficiently difficult now.

jonathanvajda committed:
Penultimate version I believe

Finally fixed all of them to the best of my ability, sans others' feedback.

jonathanvajda commented:
@giacomodecolle check this out?

jonathanvajda commented:
@JaronJCheung

jonathanvajda commented:
@emmoramor

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,389 @@
+```
+
+**Problem 7-2.**
+- List all of the furniture that has been recalled for reason of child safety. In this RDF model, all furniture has an active safety status of "Approved" or "Recalled" (they do not have a null value).
````
perhaps say something regarding how people should return the furniture (by their label)

jonathanvajda replied:
fixed!

giacomodecolle commented on Project 4/SPARQL-JV.md:
```diff
@@ -0,0 +1,389 @@
+      ?subject foaf:birthday ?birthDate ;
+            foaf:gender “male” ;
+            foaf:mbox ?mbox .
+      FILTER (?birthdate < "2007-03-27T00:00:00+05:30"^^xsd:dateTime)
```
I guess you are using the "2007-03-27T00:00:00+05:30" as the date? I can't remember what xsd:dateTime is doing. But basically you want the user to be able to replicate this solution with different dates on the basis of what day they are doing the exercise

jonathanvajda replied:
Yes, that was what I was intending, but I will change it to just an age value and filter comparing the ?age with a set integer

giacomodecolle commented on Project 4/SPARQL-JV.md:
```diff
@@ -0,0 +1,389 @@
+
+SELECT ?park
+WHERE {
+      ?park ro:located_in bfo:site ;
```
Why do you need to go through the bfo:site detour? Can't you just have ?park touronto:contains_habitat touronto:naturalHabitat

jonathanvajda replied:
This is a good question. I had in mind a worry where I was modeling overlap (a is entirely contained in b, a overlaps with b, b is entirely contained in b). Depending on the meaning of "contains_habitat" the habitat might span states/provinces. So I used "site" as a flexible middle term for scaling the size.

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,389 @@
+}
+```
+
+**Problem 7-6.**
````
This is in line with what I was thinking of doing, by specifically telling the user what properties to use

jonathanvajda replied:
Agreed!

giacomodecolle commented on Project 4/SPARQL-JV.md:
```diff
@@ -0,0 +1,389 @@
+
+SELECT ?productName ?modelValue ?releaseDate
+WHERE {
+      ?device ontocam:manufactured_by ontocam:Fujifilm ;
```
I think that when you use semicolons you should skip repeating ?device.

jonathanvajda replied:
Agreed. Fixed!

giacomodecolle commented on Project 4/SPARQL-JV.md:
```diff
@@ -0,0 +1,389 @@
+SELECT ?productName ?productModelValue ?inventoryValue ?productMSRP ?tentativeSalePrice
+WHERE {
+  ?product ro:instance_of homedev:outdoorFurniture .
+  ?product ro:has_value homedev:inventoryValue .
```
How are you counting 20 furniture models? How are we counting the one with the most inventory? I am still not comfortable with how BIND works, so this will sound ignorant, is there a reason why it is 0-8 and not 0.4?

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,389 @@
+```
+
+
+**Problem 6-1.**
````
Isn't this a bit easy compared to, say, 6.5?

jonathanvajda replied:
I added an ordering, additional criteria, and other details. I added the namespace schema guidance.

giacomodecolle commented on Project 4/SPARQL-JV.md:
```diff
@@ -0,0 +1,389 @@
+  FILTER (?function != homedev:twoShelf && 
+          ?salePriceValueUSD < 150 && 
+          ?material = homedev:woodMaterial && 
+          ?material != homedev:metalwireMaterial)
```
Food for thought: should we force "composed_primarily_of" to take at most one value?

jonathanvajda replied:
Good question. The relation was directly from the real Relations Ontology. It might be functional, and so the second is redundant?

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,389 @@
+**Problem 5-2.**
+- The federal department of education gives special grants to those educational institutions that garner the highest enrollments in the state (only one per state), whether that school is public or private. The total enrollment includes both undergraduate and postgraduate students. Find the college or university that would be eligible for such a grant in the state of California.
+- Give the name of the institution and number of students of a post-secondary educational institution located in California with the highest enrollment of students (undergrad and grad together).
+```
````
I am wondering if this is an hard 6

jonathanvajda replied:
Agreed. I ramped up the expectations with CONSTRUCT

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,389 @@
+- Display the name of the oldest 25 ziggurats, current country location, name of associated culture when it was formed, and estimated age. Order by estimated age, descending (or date of completion ascending).
+```
+PREFIX anthro: <http://anthropologywiki.org/schema/ontology/>
+
````
I wouldn't say this is a level 4, it is an interesting query but for level 4s I am thinking about things like using many different functions, creating graphs, doing math, etc. You do use order by asc and filter that are interesting but if I think of the level 4s in codewars this looks easier

giacomodecolle commented on Project 4/SPARQL-JV.md:
````diff
@@ -0,0 +1,389 @@
+**Problem 4-2.**
+- Find all public elections for president held in sovereign states that were formerly members of the USSR and not currently dissolved. Display the country and the winner as well.
+- Note: be sensitive to the open world assumption.
+```
````
This is also interesting for the use of the status dissolved. Again, not sure whether the complexity is proper for a level 4.

jonathanvajda committed:
addressing review objections

Giacomo gave feedback that this would be helpful. I think it would be helpful for all of the prompts, but I will probably not be able to furnish them all in time.

jonathanvajda committed:
fixed age query

I was making it too difficult to filter by age. Fixed!

jonathanvajda committed:
restated ?device with semicolon

Giacomo's comment was right. Fixed!

jonathanvajda committed:
6-1 too easy?

Giacomo said it seemed too easy. I agree. I think it is better now.

jonathanvajda committed:
typo on primarily_composed_of

fixed primarily composed of

jonathanvajda committed:
5-2 was too easy Giacomo says...

I don't want it to be a kata 6, so I have ramped up the expectations by requiring a CONSTRUCT

jonathanvajda committed:
Update SPARQL-JV.md

raising the number of operators or functions for my Kata-7's, per Jaron's recommendation

jonathanvajda committed:
Update SPARQL-JV.md

Hopefully I have raised the expectations
